src/threads/thread.h

```diff
@@ -80,22 +82,50 @@ typedef int tid_t;
    only because they are mutually exclusive: only a thread in the
    ready state is on the run queue, whereas only a thread in the
    blocked state is on a semaphore wait list. */
+struct tcb
+{
+  tid_t tid;
+  bool exit;
+  bool wait;
+  bool goa;
+  int exit_code;
+  char * argv;
+  char * prog;
+  struct thread * me;
+  struct thread * parent;
+  struct list_elem elem;
+  struct semaphore sema;
+  struct semaphore wait_sema;
+};
```

Comment:
Moved this comment to a more appropriate position.... 

What does TCB stand for?
A thread will hold this struct IF the thread is going to be used for a user program

Reply:
Thread Control Block (TCB) is a data structure in the operating system kernel which contains thread-specific information needed to manage it